internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -1100,3 +1100,182 @@ func TestReconcileInfrastructure(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileCertificateExpiry(t *testing.T) {
+	fakeTimeString := "2020-01-01T00:00:00Z"
+	fakeTime, _ := time.Parse(time.RFC3339, fakeTimeString)
+	fakeMetaTime := &metav1.Time{Time: fakeTime}
+
+	fakeTimeString2 := "2020-01-01T00:00:00Z"
+	fakeTime2, _ := time.Parse(time.RFC3339, fakeTimeString2)
+	fakeMetaTime2 := &metav1.Time{Time: fakeTime2}
+
+	bootstrapConfigWithExpiry := map[string]interface{}{
+		"kind":       "GenericBootstrapConfig",
+		"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
+		"metadata": map[string]interface{}{
+			"name":      "bootstrap-config-with-expiry",
+			"namespace": metav1.NamespaceDefault,
+			"annotations": map[string]interface{}{
+				clusterv1.MachineCertificatesExpiryDateAnnotation: fakeTimeString,
+			},
+		},
+		"spec": map[string]interface{}{},
+		"status": map[string]interface{}{
+			"ready":          true,
+			"dataSecretName": "secret-data",
+		},
+	}
+
+	bootstrapConfigWithoutExpiry := map[string]interface{}{
+		"kind":       "GenericBootstrapConfig",
+		"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
+		"metadata": map[string]interface{}{
+			"name":      "bootstrap-config-without-expiry",
+			"namespace": metav1.NamespaceDefault,
+		},
+		"spec": map[string]interface{}{},
+		"status": map[string]interface{}{
+			"ready":          true,
+			"dataSecretName": "secret-data",
+		},
+	}
+
+	tests := []struct {
+		name     string
+		machine  *clusterv1.Machine
+		expected func(g *WithT, m *clusterv1.Machine)
+	}{
+		{
+			name: "worker machine with certificate expiry annotation should not update expiry date",
+			machine: &clusterv1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "bootstrap-test-existing",
+					Namespace: metav1.NamespaceDefault,
+					Annotations: map[string]string{
+						clusterv1.MachineCertificatesExpiryDateAnnotation: fakeTimeString,
+					},
+				},
+				Spec: clusterv1.MachineSpec{
+					Bootstrap: clusterv1.Bootstrap{},
+				},
+			},
+			expected: func(g *WithT, m *clusterv1.Machine) {
+				g.Expect(m.Status.CertificatesExpiryDate).To(BeNil())
+			},
+		},
+		{
```

Comment:
nit: Let's add one more case where the annotation is only set on the Machine
I think then we covered all variants

Reply:
done.